### PR TITLE
feat(dashboard): render, create and edit plans

### PR DESCRIPTION
**Pull request by moalidv**

### CODE REVIEW

File: packages/luna/src/components/Form/Controller.tsx [hunk 1]
```diff
@@ -92,14 +94,14 @@ export function Input<T extends FieldValues>({
...
-export function NumericInput<T extends FieldValues>({
+export function NumericInput<T extends FieldValues, N extends FieldPath<T>>({
   control,
   name,
   rules,
   ...props
 }: {
   control: Control<T>;
-  name: Path<T>;
+  name: N;
```
Comment (neuodev) [inline packages/luna/src/components/Form/Controller.tsx:104]:
```
export function NumericInput<T extends FieldValues>({
  control,
  name,
  rules,
  ...props
}: {
  control: Control<T>;
  name: Path<T>;
```

File: packages/luna/src/locales/ar-eg.json [hunk 2]
```diff
@@ -217,6 +217,23 @@
   "invoices.process.submit.approveUpdateRequest": "قبول تحديث بيانات طلب السحب",
   "invoices.process.success": "تم تحديث بيانات طلب السحب بنجاح",
   "invoices.process.error": "لديك مشكلة في تحديث بيانات طلب السحب",
+  "dashboard.plans.title": "الخطط",
+  "dashboard.plans.createPlanBtn": "إنشاء خطة",
+  "dashboard.plan.title": "عنوان الخطة",
+  "dashboard.plan.weeklyMinutes": "الدقائق في الأسبوع",
```
Comment (neuodev) [inline packages/luna/src/locales/ar-eg.json:223]:
```suggestion
  "dashboard.plan.weeklyMinutes": "الدقائق الإسبوعية",
```

File: packages/luna/src/locales/ar-eg.json [hunk 3]
```diff
@@ -217,6 +217,23 @@
...
+  "dashboard.plan.fullMonthPrice": "السعر الشهري",
+  "dashboard.plan.fullMonthDiscount": "الخصم الشهري",
+  "dashboard.plan.fullQuarterPrice": "السعر الربع سنوي",
+  "dashboard.plan.fullQuarterDiscount": "الخصم الربع سنوي",
+  "dashboard.plan.halfYearPrice": "الخصم الربع سنوي",
+  "dashboard.plan.halfYearDiscount": "الخصم الربع سنوي",
+  "dashboard.plan.fullYearPrice": "السعر السنوي",
+  "dashboard.plan.fullYearDiscount": "الخصم السنوي",
+  "dashboard.plan.forInvitesOnly": "يمكن دعوة أشخاص",
```
Comment (neuodev) [inline packages/luna/src/locales/ar-eg.json:232]:
```suggestion
  "dashboard.plan.forInvitesOnly": "مخصصه لدعوة الاشخاص فقط",
```

**Review by neuodev — CHANGES_REQUESTED** [review]
Great @moalidv! You code is clean and meet our coding standards. Please take a look at my minor comments. Thanks in advance!

**Comment by neuodev** [comment]
image

Please truncate the table header text (check [this](https://tailwindcss.com/docs/text-overflow) and [this](https://css-tricks.com/snippets/css/truncate-string-with-ellipsis/)) and the full text should be shown in the a popover when hovering over a specific column header.

File: apps/dashboard/src/components/Plans/List.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            ]}
+          />
+        ),
+      }),
+    ],
+    [columnHelper]
+  );
+
+  if (query.isLoading) return <Loading className="h-1/4" />;
+  if (!query.data) return null;
```
[the remark this answers is not in the document]
Comment (neuodev) [reply apps/dashboard/src/components/Plans/List.tsx:131]: @moalidv please implement the error element.

**Comment by neuodev** [comment]
image

This filed should be a `Duration` input.

**Comment by neuodev** [comment]
image

Let's try making our commitment more specific. *some* is a generic description.  *implement some changes* is quite unclear. 

Please update your commitment description to be more specific. Here how you can update you commitment message in `git`

- https://kodekloud.com/blog/change-commit-message-in-git/ 
- https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d

File: apps/dashboard/src/components/Plans/List.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+  const close = useCallback(() => setPlan(null), []);
+  const columnHelper = createColumnHelper<IPlan.MappedAttributes>();
+  const columns = useMemo(
+    () => [
+      columnHelper.accessor("id", {
+        header: intl("dashboard.plans.id.header"),
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor("alias", {
+        header: intl("dashboard.plans.alias.header"),
```
Comment (neuodev) [inline apps/dashboard/src/components/Plans/List.tsx:31]:
```suggestion
        header: intl("dashboard.plans.id"),
        cell: (info) => info.getValue(),
      }),
      columnHelper.accessor("alias", {
        header: intl("dashboard.plans.alias"),
```

File: apps/dashboard/src/components/Plans/List.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        },
+      }),
+      columnHelper.accessor("active", {
+        header: intl("dashboard.plan.active"),
+        cell: (info) => {
+          return <BooleanField checked={info.getValue()} />;
+        },
+      }),
+      columnHelper.accessor("createdAt", {
+        header: intl("dashboard.plans.createdAt.header"),
```
Comment (neuodev) [inline apps/dashboard/src/components/Plans/List.tsx:96]:
```suggestion
        header: intl("dashboard.plans.createdAt"),
```

File: apps/dashboard/src/components/Plans/List.tsx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        cell: (info) => {
+          return <DateField date={info.getValue()} />;
+        },
+      }),
+      columnHelper.accessor("updatedAt", {
+        header: intl("dashboard.plans.updatedAt.header"),
```
Comment (neuodev) [inline apps/dashboard/src/components/Plans/List.tsx:102]:
```suggestion
        header: intl("dashboard.plans.updatedAt"),
```

**Review by neuodev — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/components/Plans/List.tsx (hunk 5 above)
Comment (moalidv) [reply apps/dashboard/src/components/Plans/List.tsx:31]: Done!!

File: apps/dashboard/src/components/Plans/List.tsx (hunk 6 above)
Comment (moalidv) [reply apps/dashboard/src/components/Plans/List.tsx:96]: Done!!

File: apps/dashboard/src/components/Plans/List.tsx (hunk 7 above)
Comment (moalidv) [reply apps/dashboard/src/components/Plans/List.tsx:102]: Done!

File: apps/dashboard/src/components/Plans/List.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (moalidv) [reply apps/dashboard/src/components/Plans/List.tsx:136]: > Please handle the error state as well. Similar to what we are doing [here](https://github.com/litespace-org/litespace/blob/b0dd38c003da73eea86f78677a03dd8cb0458943/apps/nova/src/pages/Lessons.tsx#L52)

image

File: apps/dashboard/package.json [hunk 8]
```diff
@@ -28,7 +29,8 @@
     "react-router-dom": "^6.23.1",
     "react-toastify": "^10.0.5",
     "redux-persist": "^6.0.0",
-    "tailwind-scrollbar": "^3.1.0"
+    "tailwind-scrollbar": "^3.1.0",
+    "@litespace/sol": "workspace:^"
```
[the remark this answers is not in the document]
Comment (moalidv) [reply apps/dashboard/package.json]: Done!

File: apps/dashboard/src/App.tsx [hunk 9]
```diff
@@ -5,6 +5,7 @@ import { ErrorPage } from "@litespace/luna";
 import Invoices from "@/pages/Invoices";
 import Media from "@/pages/Media";
 import Login from "@/pages/Login";
+import Plans from "./pages/Plans";
```
[the remark this answers is not in the document]
Comment (moalidv) [reply apps/dashboard/src/App.tsx]: Done!

File: apps/dashboard/src/components/Plans/List.tsx [hunk 10]
```diff
@@ -0,0 +1,143 @@
+import Price from "@/components/Plans/Price";
+import Table from "@/components/Plans/Table";
+import BooleanField from "@/components/common/BooleanField";
+import DateField from "@/components/common/DateField";
+import { ActionsMenu, Loading } from "@litespace/luna";
+import { IPlan } from "@litespace/types";
+import { UseQueryResult } from "@tanstack/react-query";
+import { createColumnHelper } from "@tanstack/react-table";
+import React, { useMemo, useState, useCallback } from "react";
+import PlanForm from "./PlanForm";
```
[the remark this answers is not in the document]
Comment (moalidv) [reply apps/dashboard/src/components/Plans/List.tsx]: Done!

File: apps/dashboard/src/components/Plans/List.tsx [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+
+const List: React.FC<{
+  query: UseQueryResult<IPlan.MappedAttributes[], Error>;
+}> = ({ query }) => {
+  const [plan, setPlan] = useState<IPlan.MappedAttributes | null>(null);
+  const close = useCallback(() => {
+    setPlan(null);
+  }, []);
```
[the remark this answers is not in the document]
Comment (moalidv) [reply apps/dashboard/src/components/Plans/List.tsx]: Done!

File: apps/dashboard/src/components/Plans/List.tsx [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+  const columnHelper = createColumnHelper<IPlan.MappedAttributes>();
+  const columns = useMemo(
+    () => [
+      columnHelper.accessor("id", {
+        header: "id",
```
[the remark this answers is not in the document]
Comment (moalidv) [reply apps/dashboard/src/components/Plans/List.tsx]: Done!

File: apps/dashboard/src/components/Plans/PlanForm.tsx [hunk 13]
```diff
@@ -0,0 +1,326 @@
...
+
+  return (
+    <Dialog
+      open={open}
+      title={intl("dashboard.plan.form.create")}
+      close={onClose}
+    >
+      <Form
+        onSubmit={form.handleSubmit(onSubmit)}
+        className="min-w-96 max-h-[32rem]  overflow-y-auto scrollbar-thin scrollbar-thumb-border-stronger scrollbar-track-surface-300"
```
[the remark this answers is not in the document]
Comment (moalidv) [reply apps/dashboard/src/components/Plans/PlanForm.tsx:125]: Done!

File: apps/dashboard/src/components/Plans/PlanForm.tsx [hunk 14]
```diff
@@ -0,0 +1,326 @@
...
+                value={form.watch("alias")}
+                rules={aliasRules}
+              />
+            }
+          />
+          <Field
+            className="mb-1"
+            label={<Label>{intl("dashboard.plan.weeklyMinutes")}</Label>}
+            field={
+              <Controller.NumericInput
```
[the remark this answers is not in the document]
Comment (moalidv) [reply apps/dashboard/src/components/Plans/PlanForm.tsx]: Done!

File: packages/luna/src/locales/ar-eg.json (hunk 2 above)
Comment (moalidv) [reply packages/luna/src/locales/ar-eg.json]: Done!

**Review by neuodev — APPROVED** [review]

**Pull request merged**